connect.go

```diff
@@ -547,3 +593,147 @@ func (pm *ProducersMap) unsetStationProducers(stationName string) {
 		}
 	}
 }
+func (cm *ConsumersMap) getConsumer(key string) *Consumer {
+	if (*cm) != nil && (*cm)[key] != nil {
+		return (*cm)[key]
+	}
+	return nil
+}
+
+func (cm *ConsumersMap) setConsumer(c *Consumer) {
+	internalStationName := getInternalName(c.stationName)
+	cn := fmt.Sprintf("%s_%s", internalStationName, c.realName)
+	if cm.getConsumer(cn) != nil {
+		return
+	}
+	(*cm)[cn] = c
+}
+
+func (cm *ConsumersMap) unsetConsumer(key string) {
+	delete(*cm, key)
+}
+
+func (cm *ConsumersMap) unsetStationConsumers(stationName string) {
+	internalStationName := getInternalName(stationName)
+	for k, v := range *cm {
+		intetnalStationV := getInternalName(v.stationName)
+		if intetnalStationV == internalStationName {
+			cm.unsetConsumer(k)
+		}
+	}
+}
+
+// FetchMessages - Consume a batch of messages.
+func (c *Conn) FetchMessages(stationName string, consumerName string, opts ...FetchOpt) ([]*Msg, error) {
+	var consumer *Consumer
+	cm := c.getConsumersMap()
+	internalStationName := getInternalName(strings.ToLower(stationName))
+	cons := cm.getConsumer(fmt.Sprintf("%s_%s", internalStationName, strings.ToLower(consumerName)))
+	defaultOpts := getDefaultFetchOptions()
+	defaultOpts.ConsumerName = consumerName
+	defaultOpts.StationName = stationName
+	for _, opt := range opts {
+		if opt != nil {
+			if err := opt(&defaultOpts); err != nil {
+				return nil, memphisError(err)
+			}
+		}
+	}
+	if cons == nil {
+		if defaultOpts.GenUniqueSuffix {
+			co, err := c.CreateConsumer(stationName, consumerName, BatchMaxWaitTime(defaultOpts.BatchMaxTimeToWait), BatchSize(defaultOpts.BatchSize), ConsumerGroup(defaultOpts.ConsumerGroup), ConsumerErrorHandler(defaultOpts.ErrHandler), LastMessages(defaultOpts.LastMessages), MaxAckTime(defaultOpts.MaxAckTime), MaxMsgDeliveries(defaultOpts.MaxMsgDeliveries), StartConsumeFromSequence(defaultOpts.StartConsumeFromSequence), ConsumerGenUniqueSuffix())
+			if err != nil {
+				return nil, err
+			}
+			consumer = co
+		} else {
+			con, err := c.CreateConsumer(stationName, consumerName, BatchMaxWaitTime(defaultOpts.BatchMaxTimeToWait), BatchSize(defaultOpts.BatchSize), ConsumerGroup(defaultOpts.ConsumerGroup), ConsumerErrorHandler(defaultOpts.ErrHandler), LastMessages(defaultOpts.LastMessages), MaxAckTime(defaultOpts.MaxAckTime), MaxMsgDeliveries(defaultOpts.MaxMsgDeliveries), StartConsumeFromSequence(defaultOpts.StartConsumeFromSequence))
+			if err != nil {
+				return nil, err
+			}
+			consumer = con
+		}
+	} else {
+		consumer = cons
+	}
+	msgs, err := consumer.Fetch(defaultOpts.BatchSize)
+	if err != nil {
+		return nil, err
+	}
+	return msgs, nil
+}
+
+// ConsumerName - name for the consumer.
+func FetchConsumerName(name string) FetchOpt {
```

Comment:
I've commented that we don't need it since it is a mandatory field why have you left it?